### PR TITLE
`no-for-loop`: Ignore known non-array loop variables

### LINKED ISSUE #738: `no-for-loop` rule produces invalid autofix in trying to use `.entries()` to iterate a string
In situations like this where we can tell that a variable being iterated is a string (not an array), we should not convert the loop to use the `.entries()` function which does not exist on strings.

Before:
```js
const myString = 'helloworld';
for (let i = 0; i < myString.length; i++) {
    console.log(i, myString[i])
}
```

After autofix (invalid):
```js
const myString = 'helloworld';
for (const [i, element] of myString.entries()) {
    console.log(i, element);
}
```

### PR DESCRIPTION
**Pull request by bmish**
In situations like this where we can tell that a variable being iterated is not an array (i.e. a string), we should not convert the loop to use the `.entries()` function which would not exist on non-arrays.

Fixes #738.

### CODE REVIEW

**Review by fisker — APPROVED** [review]
LGTM

**Comment by bmish** [comment]
Force-pushed to update commit author, no code change.

**Comment by sindresorhus** [comment]
@bmish Can you fix the merge conflict?

**Comment by bmish** [comment]
> @bmish Can you fix the merge conflict?

Fixed.

File: rules/no-for-loop.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const isLiteralValue = require('./utils/is-literal-value');
 const avoidCapture = require('./utils/avoid-capture');
 const getChildScopesRecursive = require('./utils/get-child-scopes-recursive');
 const singular = require('./utils/singular');
+const {getStaticValue} = require('eslint-utils');
```
Comment (sindresorhus) [inline rules/no-for-loop.js:8]: See the first require statement.

Comment (bmish) [reply rules/no-for-loop.js]: Fixed. I'm used to the [import/no-duplicates](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-duplicates.md) rule catching/autofixing that for me in some of my own codebases.

**Pull request merged**